src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1241,4 +1240,38 @@ public static void onEntityEnterSection(Entity entity, long packedOldPos, long p
         MinecraftForge.EVENT_BUS.post(new EntityEvent.EnteringSection(entity, packedOldPos, packedNewPos));
     }
 
+    @Nullable
+    public static MutableComponent getJoinMessage(Player player, String cachedName)
+    {
+        PlayerEvent.JoinMessageFormat event = new PlayerEvent.JoinMessageFormat(player);
+        MinecraftForge.EVENT_BUS.post(event);
+
+        if(event.isCanceled()) return null;
+
+        MutableComponent mutablecomponent = event.getJoinMessage();
+        if(mutablecomponent == null) {
+            if (cachedName == null || player.getGameProfile().getName().equalsIgnoreCase(cachedName)) {
+                mutablecomponent = new TranslatableComponent("multiplayer.player.joined", player.getDisplayName());
+            } else {
+                mutablecomponent = new TranslatableComponent("multiplayer.player.joined.renamed", player.getDisplayName(), cachedName);
+            }
```

Comment:
This could probably be reduced to a ternary operator, but since it's just a movement of vanilla code, this is more of an optional thing.